src/libraries/System.Net.Security/src/System/Net/Security/SslStreamCertificateContext.Linux.cs

```diff
@@ -26,13 +26,21 @@ public partial class SslStreamCertificateContext
         private byte[]? _ocspResponse;
         private DateTimeOffset _ocspExpiration;
         private DateTimeOffset _nextDownload;
+        // Private copy of the intermediate certificates, in case the user decides to dispose the
+        // instances reachable through IntermediateCertificates property.
+        private X509Certificate2[] _privateIntermediateCertificates;
         private Task<byte[]?>? _pendingDownload;
         private List<string>? _ocspUrls;
-        private X509Certificate2? _ca;
 
         private SslStreamCertificateContext(X509Certificate2 target, ReadOnlyCollection<X509Certificate2> intermediates, SslCertificateTrust? trust)
         {
             IntermediateCertificates = intermediates;
+            _privateIntermediateCertificates = new X509Certificate2[intermediates.Count];
+            for (int i = 0; i < intermediates.Count; i++)
+            {
+                _privateIntermediateCertificates[i] = new X509Certificate2(intermediates[i]);
+            }
+
```

Comment:
Short chains (self signed, or root->end-entity direct) have an intermediates.Count of 0, which is dealt with later on.  Here it could avoid the unique empty array if it wanted... though the number of these objects created per process is fairly small in the grand scheme of things.

```suggestion
            if (intermediates.Count > 0)
            {
                _privateIntermediateCertificates = new X509Certificate2[intermediates.Count];

                for (int i = 0; i < intermediates.Count; i++)
                {
                    _privateIntermediateCertificates[i] = new X509Certificate2(intermediates[i]);
                }
            }
            else
            {
                _privateIntermediateCertificates = Array.Empty<X509Certificate2>();
            }

```